ui/src/app/components/extend-initial-credits-modal.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import { useState } from 'react';
+
+import { Profile } from 'generated/fetch';
+
+import { Button, StyledExternalLink } from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import { Spinner } from 'app/components/spinners';
+import { profileApi } from 'app/services/swagger-fetch-clients';
+import colors from 'app/styles/colors';
+import { displayDateWithoutHours, minusDays } from 'app/utils/dates';
+import { notificationStore, serverConfigStore } from 'app/utils/stores';
+import { supportUrls } from 'app/utils/zendesk';
+
+import { AoU } from './text-wrappers';
+
+interface Props {
+  onClose: Function;
+  expirationDate: number;
+}
+
+const ExtensionDescription = ({ expirationDate }) => {
+  const {
+    initialCreditsValidityPeriodDays,
+    initialCreditsExpirationWarningDays,
+  } = serverConfigStore.get().config;
+  return (
+    <div>
+      Initial credits provided by the <AoU /> Research Program expire after{' '}
+      {initialCreditsValidityPeriodDays} days. You are eligible to request an
+      extension after{' '}
+      {displayDateWithoutHours(
+        minusDays(expirationDate, initialCreditsExpirationWarningDays)
+      )}{' '}
+      which is {initialCreditsExpirationWarningDays} days before your current
+      expiration date {displayDateWithoutHours(expirationDate)}.
+      <StyledExternalLink
+        href={supportUrls.initialCredits}
+        style={{ marginTop: '1rem' }}
+      >
+        Learn more about credits & setting up a billing account
+      </StyledExternalLink>
+    </div>
+  );
+};
+
+export const ExtendInitialCreditsModal = ({
+  onClose,
+  expirationDate,
+}: Props) => {
+  const [saving, setSaving] = useState(false); // saving refers to the loading request time
```

Comment:
I actually prefer this without the comment.  setting a verb-ing state while we are verb-ing is a common pattern for us, so I don't feel it needs explanation.

But now I'm wondering what "loading request time" is, and why this isn't `loading` instead.  Actually... maybe this should be `extending`?

Reply:
I don't think that comment makes sense anyways  😅

I changed it to `extending`. This variable is intended to prevent the user from requesting an extension multiple times. It locks the interface until the extension requests returns.